src/composables/auth.ts

```diff
@@ -70,33 +76,44 @@ export const useAuth = createCustomScopedComposable(() => {
       }).then(() => {
         isAuthenticated.value = true;
       });
-    } catch (error) {
-      return Promise.reject(error);
+    } if (!IS_MOBILE_APP) {
+      return authenticateWithPassword(password!).then(({ encryptionKey: newEncryptionKey }) => {
+        setEncryptionKey(newEncryptionKey);
+        isAuthenticated.value = true;
+      });
     }
+    return Promise.resolve();
   }
 
-  function logout() {
+  /**
+   * Checks if the user should be kept logged in based on the current session.
+   */
+  async function shouldKeepExtensionLoggedIn() {
+    return !!(await getSessionEncryptionKey());
+  }
+
+  async function logout() {
```

Comment:
I checked and every time we run `logout()` the next thing is `openSecureLoginModal()`. Maybe these 2 things can be combined?
What do you think @CedrikNikita ?

Reply:
What about an extra function called something like `relog` which would call those functions?
I'm not sure about merging the functionality of those functions though. I'd like to keep the separate functions as well

Reply:
Here I created separate method for this (`lockWallet`): https://github.com/superhero-com/superhero-wallet/pull/3275/files#diff-dc9f684daabd265d4a2b08dfc8537c4f7925f66dae34b1859945a1fd889b3635R140
Can you take a look?